src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509Pal.Windows.PublicKey.cs

```diff
@@ -98,10 +98,7 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
                     }
 
                     keyBlob = ExportKeyBlob(bCryptKeyHandle, blobFormat);
-                    using (CngKey cngKey = CngKey.Import(keyBlob, blobFormat))
-                    {
-                        key = factory(cngKey);
-                    }
+                    key = factory(CngKey.Import(keyBlob, blobFormat));
```

Comment:
I'm probably missing something obvious, but "And also leaving some of the originals for finalization." what was being left behind for finalization, since the `cngKey` was in a `using`?

To me it seems like previously we were importing the key, duplicating, and then disposing the original. Now it seems like we are importing the key, but with no duplication and no disposal.

It's better, I just want to understand what was behind left behind for finalization.

Reply:
> but "And also leaving some of the originals for finalization." what was being left behind for finalization, since the cngKey was in a using?

e.g.
https://github.com/dotnet/runtime/pull/72120/files#diff-cd2d7c549b4812687e36c74fa769c8dcbe7ce42fcdfe6bfb2fa41f7aa643aaeeL60

Reply:
Right. Yep. Missed something obvious.